core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -616,8 +618,21 @@ protected Builder buildAggregate(Aggregate e, Builder builder,
       } else if (aggCall.getAggregation() instanceof SqlMinMaxAggFunction) {
         aggCallSqlNode = dialect.rewriteMaxMinExpr(aggCallSqlNode, aggCallRelDataType);
       }
+      isCountStar =  (RelOptUtil.hasCalcViewHint(e) && selectList.isEmpty() && aggCall.getAggregation().getKind() == SqlKind.COUNT);
       addSelect(selectList, aggCallSqlNode, e.getRowType());
     }
+    if (isCountStar) {
```

Comment:
Should we also do this only in case of a CALCULATION_VIEW?

Reply:
I will check, but I'd say yes